src/transformers/trainer.py

```diff
@@ -1315,8 +1315,6 @@ def prediction_loop(
                 label_ids = xm.mesh_reduce("eval_label_ids", label_ids, torch.cat)
             if eval_losses is not None:
                 eval_losses = xm.mesh_reduce("eval_losses", torch.tensor(eval_losses), torch.cat).tolist()
-            if samples_count is not None:
```

Comment:
Is this intentional?

Reply:
Yes, removed artefact from FLOPS PR (this runs on TPU and wasn't caught by the test).